src/backends/unix/system.c

```diff
@@ -548,27 +548,92 @@ Sys_IsFile(const char *path)
 	return false;
 }
 
-char *
-Sys_GetHomeDir(void)
+#ifdef USE_XDG
+static char *
+GetXDGPath(const char *xdg)
 {
-	static char gdir[MAX_OSPATH];
-	char *home;
+	char* buffer = calloc(MAX_OSPATH, sizeof(char));
+	if (!buffer) {
+		goto fail;
+	}
 
-	home = getenv("HOME");
+	const char* env = getenv(xdg);
+	const char* fmt = "%s/%s/";
 
-	if (!home)
-	{
-		return NULL;
+	if (!env) {
+		env = getenv("HOME");
+		if (!env) {
+			goto fail;
```

Comment:
Code could free buffer and return NULL as it's single place of error, goto redundant.